cishouseholds/pipeline/run.py

```diff
@@ -33,7 +34,9 @@ def run_from_config():
         print(f"Spark UI: {get_spark_ui_url()}")  # functional
         print(f"Spark application ID: {get_spark_application_id()}")  # functional
         pipeline_stage_list = [stage for stage in config["stages"] if stage.pop("run")]
-        pipeline_error_count = run_pipeline_stages(pipeline_stage_list, run_id)
+        pipeline_error_count = run_pipeline_stages(
+            pipeline_stage_list, run_id, config["retry_n_times_on_fail"], config["retry_wait_time"]
```

Comment:
I'd drop the n from here, or change to "number" if necessary

```suggestion
            pipeline_stage_list, run_id, config["retry_times_on_fail"], config["retry_wait_time"]
```

Reply:
Also, both of these should have defaults set here (I'd suggest 0), for backwards compatibility
You can use `dict.get("name", default)` to set this